validation/superchain-genesis_test.go

```diff
@@ -0,0 +1,49 @@
+package validation
+
+import (
+	"testing"
+
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/stretchr/testify/require"
+)
+
+func testGenesisHashOfChain(t *testing.T, chainID uint64) {
+	chainConfig, ok := OPChains[chainID]
+	if !ok {
+		t.Fatalf("no chain with ID %d found", chainID)
+	}
+
+	declaredGenesisHash := chainConfig.Genesis.L2.Hash
+
+	// In this step, we call a function from op-geth which utilises
+	// superchain.OPChains, superchain.LoadGenesis, superchain.LoadContractBytecode
+	// to reconstruct a core.Genesis and compute its block hash.
+	// We can then compare that against the declared genesis block hash.
+	// TODO core.LoadOPStackGenesis actually already performs the check we are about to do
+	// TODO core.LoadOPStackGenesis could be moved to superchain
```

Comment:
Placeholder for further work I would like to do, possibly on this PR, after getting some feedback.

Reply:
Let's start with getting the tests working, then we can refactor.